### PR TITLE
Converted `BaSiC` to a pydantic class

### LINKED ISSUE #18: Create a pydantic settings class
The current development branch implements a `Settings` class that holds settings for the `BaSiC` model class. This task should do one of two things:

1. Turn `BaSiC`  class in a `pydantic.BaseModel` subclass that implements everything in `Settings`, then remove the `Settings` class.
2. Turn `Settings` into a `pydantic.BaseModel` subclass

I vote for option 1, since the utility functions of `pydantic.BaseModel` make housing the settings for `BaSiC` in a separate class arbitrary. This will help reduce the complexity of the code since we expect the inputs to `BaSiC` to contain all parameters for `Settings`.

In addition, ensure `pydantic` is added to `setup.cfg`.

### PR DESCRIPTION
**Pull request by Nicholas-Schaub**
### Summary
This PR makes two major changes:
1. Converts `BaSiC` to a pydantic class.
2. Removes `settings.py`, since `BaSiC` now contains all settings

### Details
In addition, I made some design choices to change some model settings. These include:
1. Move `timelapse` as a setting parameter for the model to an input parameter for `predict`. This is in line to our discussions on the behavior of baseline/photobleaching calculation.
2. Change `darkfield` to `get_darkfield` 
3. Add `flatfield` and `darkfield` as `np.ndarray` parameters. These will hold the flatfield and darkfield components calculated using `fit`. This should be discussed, since it would remove the necessity for the `Profile` class.

In addition to this, very simple unit tests were created to confirm that the base configuration settings for `BaSiC` are valid.

Resolves #18

### CODE REVIEW

File: src/pybasic/pybasic.py [hunk 1]
```diff
@@ -24,79 +25,104 @@
...
+    estimation_mode: EstimationMode = Field(
+        "l0",
+        description="Flatfield offset for weight updates.",
+    )
+    flatfield: np.ndarray = Field(
+        default_factory=lambda: np.zeros((128, 128), dtype=np.float64),
+        description="Holds the flatfield component for the shading model.",
+        exclude=True,  # Don't dump to output json/yaml
+    )
+    get_dark: bool = Field(
```
Comment (tdmorello) [inline src/pybasic/pybasic.py:67]: I think this should be `get_darfield`

File: src/pybasic/pybasic.py [hunk 2]
```diff
@@ -118,19 +144,14 @@ def fit(self, images: Iterable[ArrayLike]):
...
-
-        Args:
-            im_stack: input images as a stack
-        """
-        ...  # do stuff and set `self.params`
-
-    def predict(self, images: Iterable[np.ndarray]) -> List[np.ndarray]:
+    def predict(
+        self, images: np.ndarray, timelapse: bool = False
+    ) -> Union[Tuple[np.ndarray, np.ndarray], np.ndarray]:
```
Comment (tdmorello) [inline src/pybasic/pybasic.py:149]: Change return type to `np.ndarray`?

Comment (tdmorello) [reply src/pybasic/pybasic.py:149]: User can grab `flatfield` and `darkfield` separately.

File: src/pybasic/pybasic.py [hunk 3]
```diff
@@ -141,20 +162,27 @@ def predict(self, images: Iterable[np.ndarray]) -> List[np.ndarray]:
...
+        output = apply_profiles(images)
+
+        return output
 
     # REFACTOR large datasets will probably prefer saving corrected images to
     # files directly, a generator may be handy
-    def fit_predict(self, images: Iterable[ArrayLike]) -> List[np.ndarray]:
+    def fit_predict(
+        self, images: ArrayLike, timelapse: bool = True
+    ) -> Union[Tuple[np.ndarray, np.ndarray], np.ndarray]:
```
Comment (tdmorello) [inline src/pybasic/pybasic.py:185]: Same comment about return type

File: src/pybasic/pybasic.py [hunk 4]
```diff
@@ -167,30 +195,20 @@ def fit_predict(self, images: Iterable[ArrayLike]) -> List[np.ndarray]:
             >>> profiles, corrected = basic.fit_predict(images)
         """
         self.fit(images)
-        corrected = self.predict(images)
-        return corrected
+        corrected = self.predict(images, timelapse)
```
Comment (tdmorello) [inline src/pybasic/pybasic.py:198]: `return corrected` should be added back or `return self.predict(images, timelapse)`

Comment (tdmorello) [inline src/pybasic/pybasic.py:198]: `self.corrected` ? Or `return self.predict(...`

**Review by tdmorello — COMMENTED** [review]
Some comments about return types of `predict` and `fit_predict`.

File: src/pybasic/pybasic.py (hunk 2 above)
Comment (Nicholas-Schaub) [reply src/pybasic/pybasic.py:149]: Per #14 , we should return
1. `np.ndarray` when `timelapse=False`
2. `(np.ndarray,np.ndarray)` when `timelapse=True`

The first array should be the output images, the second array should be the baseline offsets. This should be made more clear in the documentation. I just put these here as placeholders.

File: src/pybasic/pybasic.py (hunk 3 above)
Comment (Nicholas-Schaub) [reply src/pybasic/pybasic.py:185]: Same comment as above :)

File: src/pybasic/pybasic.py (hunk 2 above)
Comment (tdmorello) [reply src/pybasic/pybasic.py:149]: Ohh, right!

**Review by tdmorello — APPROVED** [review]

**Pull request merged**